test/Microsoft.Extensions.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -36,6 +36,17 @@ public void GetRequiredService_Throws_WhenNoServiceRegistered()
         }
 
         [Fact]
+        public void IRequiredServiceAwareService_GetRequiredService_Throws_WhenNoServiceRegistered()
+        {
+            // Arrange
+            var serviceProvider = new RequiredServiceAwareServiceProvider();
+
+            // Act + Assert
+            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredService<IFoo>(),
```

Comment:
Could you throw a different exception here? It's hard to tell if the exception comes from your service provider \ from the extension method.